src/Core/Managed/Shared/Extensibility/Implementation/BroadcastProcessor.cs

```diff
@@ -0,0 +1,148 @@
+﻿namespace Microsoft.ApplicationInsights.Shared.Extensibility.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    /// <summary>
+    /// An <see cref="ITelemetryProcessor"/> that asynchronously sends the data to multiple child telemetry processors.
+    /// </summary>
+    internal class BroadcastProcessor : ITelemetryProcessor, IDisposable
```

Comment:
``` csharp
foreach(var sink in sinks) {
  try {
    sink.Process(telemetry); 
  } catch (Exception e) {
    // Process error
  }
}
```
>Problem: sinks invoked one at a time, making the pipeline slower than it could be

If every sink implementation just do what broadcast thing does - put item in queue of sort and return the control back, performance will be the same, will it? This makes code much easier to understand and follow. And opens so many possibilities. Like if you have a sampling processor that samples very aggressively it may be more efficient to sample before scheduling task. You will save a lot of overhead of scheduler by queuing 1/100th of all telemetry